src/main/java/net/dv8tion/jda/core/requests/restaction/order/CategoryOrderAction.java

```diff
@@ -0,0 +1,152 @@
+        this.category = category;
+        this.type = type;
+
+        // Add all channels of a type in this Category
+        switch(type)
+        {
+            case TEXT:
+                this.orderList.addAll((List) category.getTextChannels());
+                break;
+            case VOICE:
+                this.orderList.addAll((List) category.getVoiceChannels());
+                break;
+            // Categories cannot currently be nested within categories so we just default here.
+            // If discord does allow more types of channels to be nested withing categories in
+            // the future, adding this functionality is just a matter of adding an additional case.
+            default:
+                throw new IllegalArgumentException("Cannot order channel type: "+type.name());
+        }
+    }
+
+    /**
+     * Gets the {@link net.dv8tion.jda.core.entities.Guild Guild} that
+     * the {@link net.dv8tion.jda.core.entities.Category Category}
+     * is from.
+     *
+     * @return The {@link net.dv8tion.jda.core.entities.Guild Guild} that
+     *         the {@link net.dv8tion.jda.core.entities.Category Category}
+     *         is from.
+     */
+    public Guild getGuild()
+    {
+        return getCategory().getGuild();
+    }
+
+    /**
+     * Gets the {@link net.dv8tion.jda.core.entities.Category Category}
+     * controlled by this CategoryOrderAction.
+     *
+     * @return The {@link net.dv8tion.jda.core.entities.Category Category}
+     *         of this CategoryOrderAction.
+     */
+    public Category getCategory()
+    {
+        return category;
+    }
+
+    /**
+     * The {@link net.dv8tion.jda.core.entities.ChannelType ChannelType} of
+     * the channels that are ordered by this CategoryOrderAction.
+     *
+     * @return The {@link net.dv8tion.jda.core.entities.ChannelType ChannelType}
+     *         of the channels ordered by this CategoryOrderAction.
+     */
+    public ChannelType getChannelType()
+    {
+        return type;
+    }
+
+    @Override
+    protected RequestBody finalizeData()
+    {
+        final Member self = category.getGuild().getSelfMember();
+
+        // Make sure we have permission to actually do this
+        if(!self.hasPermission(category, Permission.MANAGE_CHANNEL))
+            throw new InsufficientPermissionException(Permission.MANAGE_CHANNEL);
+
+        JSONArray array = new JSONArray();
+        for(int i = 0; i < orderList.size(); i++)
+        {
+            JSONObject object = new JSONObject().put("id", orderList.get(i).getId())
+                                                .put("position", i);
+            array.put(object);
+        }
+
+        return getRequestBody(array);
+    }
+
+    @Override
+    protected void validateInput(T entity)
+    {
+        Checks.check(entity.getGuild().equals(category.getGuild()), "Provided channel is not from this Guild!");
```

Comment:
Instead of checking for the guild it makes more sense to check for the parent category, I'd argue.

Reply:
Agreed

Reply:
I can do this as soon as I have access to my computer